pkg/okteto/context.go

```diff
@@ -343,7 +343,27 @@ func AddOktetoCredentialsToCfg(cfg *clientcmdapi.Config, cred *types.Credential,
 	if !ok {
 		user = clientcmdapi.NewAuthInfo()
 	}
-	user.Token = cred.Token
+
+	hasKubeToken, err := hasKubeTokenCapabily(oktetoURL)
+	if err != nil {
+		return err
+	}
+	if hasKubeToken {
+		user.Token = ""
+		user.Exec = &clientcmdapi.ExecConfig{
+			Command:            "okteto",
+			Args:               []string{"kubetoken"},
+			APIVersion:         "client.authentication.k8s.io/v1",
+			InstallHint:        "Okteto needs to be installed and in your PATH to use this context. Please visit https://www.okteto.com/docs/getting-started/ for more information.",
+			ProvideClusterInfo: true,
+			InteractiveMode:    "IfAvailable",
+		}
+	} else {
+		// fallback for okteto API before client authentication support
+		// TODO: remove once we stop supporting token based authentication
+		user.Token = cred.Token
+		user.Exec = nil
+	}
```

Comment:
All of this logic will simplify a lot once we deprecate token based auth.

There's also the option of filling both `Token` and `Exec` and kubectl will default to `Token` that would also simplify all this logic. I didn't opt for this approach because it has the downside of only using `Exec` when `Token` is not set, hence we'll use this behavior only once we deprecate the `Token`. To get early feedback on this feature and manage possible errors with time (before k8s 1.26) I think this is the best approach

Reply:
Makes sense to do it in this way